addons/point_of_sale/models/res_users.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.multi
+    def unlink(self):
+        for user in self:
+            if (self.env['pos.session'].search_count([('user_id', '=', user.id), ('state', '!=', 'closed')]) 
```

Comment:
try to batch your search, you can use `self.ids` and make one global search instead of iterating on each record